Platform/ApollolakeBoardPkg/Library/Stage2BoardInitLib/Stage2BoardInitLib.c

```diff
@@ -627,6 +627,44 @@ PlatformCpuInit (
   AsmWriteMsr64 (EFI_MSR_POWER_MISC, Data);
 }
 
+/**
+  Update current boot Payload ID.
+
+**/
+VOID
+UpdatePayloadId (
+  VOID
+  )
+{
+  GEN_CFG_DATA       *GenericCfgData;
+  PLATFORM_CFG_DATA  *PlatCfgData;
+  UINT32              PayloadId;
+  UINT32              GpioLevel;
+
+  PayloadId = 0;
+  GenericCfgData = (GEN_CFG_DATA *)FindConfigDataByTag (CDATA_GEN_TAG);
+  if (GenericCfgData != NULL) {
+    if (GenericCfgData->PayloadId != AUTO_PAYLOAD_ID_SIGNATURE) {
+      PayloadId = GenericCfgData->PayloadId;
+    }
+  }
+
+  GpioLevel = 1;
+  PlatCfgData = (PLATFORM_CFG_DATA *)FindConfigDataByTag (CDATA_PLATFORM_TAG);
+  if ((PlatCfgData != NULL) && (PlatCfgData->PayloadSelGpio.Enable != 0)) {
+    // The default GPIOSet to Pull Up 20K
+    GpioGetInputValue (PlatCfgData->PayloadSelGpio.PadInfo, 0x0C, &GpioLevel);
+    if (GpioLevel == 0) {
+      PayloadId = 0;
+    } else {
+      if ((GenericCfgData != NULL) && (GenericCfgData->PayloadId == AUTO_PAYLOAD_ID_SIGNATURE)) {
```

Comment:
It will be more clear if moving line 660 before line 653. 
"Set PayloadId by Gpio pin" is required only when PayloadId is auto.
In current logic, if PayloadId is set to auto, but config data CDATA_PLATFORM_TAG is not found, the final PayloadId would be auto, this is not expected.

Reply:
In the case you mentioned,  the payload ID will be 0.

Reply:
oh, I had thought PayloadId will be auto. It is OK.